core/trino-main/src/main/java/io/trino/likematcher/LikeMatcher.java

```diff
@@ -62,6 +63,25 @@ public Optional<Character> getEscape()
         return escape;
     }
 
+    @Override
+    public boolean equals(Object o)
+    {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        LikeMatcher that = (LikeMatcher) o;
+        return Objects.equals(pattern, that.pattern) && Objects.equals(escape, that.escape);
```

Comment:
why ignore minSize, maxSize, prefix, suffix and (opaque) matcher in equals?

Reply:
The other fields appear to be derived from `pattern` and `escape` (`static LikeMatcher compile(String pattern, Optional<Character> escape)`). But I'm not too familiar with this code, so let me know if this is incorrect.

Reply:
if this is so, documenting this assumption in the code is a must.

However, compile takes `compile(String pattern, Optional<Character> escape, boolean optimize)`.
Other fields seem to be derived from `pattern`, `escape` and `optimize`.

Reply:
`optimize` seems to affect choice of Matcher implementation, not the results of matching, and as far as I can tell the choice of `optimise` seems to be deterministic in LikeFunctions.
I'll wait for @martint input here.

Reply:
if LikeMatcher equals is not purely value-based, then the equality semantics need to be clearly documented.

the optimize affects LikeMatcher behavior (supposedly the performance aspects), so in general-purpose equals should not be ignored.